Alexey-budgetCalc/modul.py

```diff
@@ -0,0 +1,211 @@
+                    }
+                )
+
+    def sort_by_category(self, category):
+        self.update()
+        category_dict = {}
+        if not (category in self.categories_set):
+            print("There is no such category as ", category)
+        else:
+            for index in range(len(self.categories_column)):
+                if self.categories_column[index] == category:
+                    if self.date_column[index] in category_dict:
+                        category_dict[self.date_column[index]] += self.expencies_column[
+                            index
+                        ]
+                    else:
+                        category_dict[self.date_column[index]] = self.expencies_column[
+                            index
+                        ]
+        return category_dict
+
+    def sort_by_categories(self):
+        print("How many categories you want to choose")
+        number = int_pos_input()
+        categories_dict = {}
+        for index in range(number):
+            category = input("Enter the category\n")
+            categories_dict[category] = self.sort_by_category(category=category)
+        return categories_dict
+
+    def sort_by_month(self):
+        self.update()
+        month = month_input()
+        month_dict = {}
+        if month in self.date_column:
+            print("There is no such month as ", month)
+        else:
+            for index in range(len(self.date_column)):
+                if month in self.date_column[index]:
+                    if self.categories_column[index] in month_dict:
+                        month_dict[
+                            self.categories_column[index]
+                        ] += self.expencies_column[index]
+                    else:
+                        month_dict[
+                            self.categories_column[index]
+                        ] = self.expencies_column[index]
+        return month_dict
+
+    def sort_by_few(self):
+        self.update()
+        month = month_input()
+        month_dict = {}
+        categories_dict = {}
+        if month in self.date_column:
+            print("There is no such month as ", month)
+        else:
+            categories_dict = self.sort_by_categories()
+            for category in categories_dict.keys():
+                for date in categories_dict[category].keys():
+                    if month in date:
+                        if date in month_dict:
+                            month_dict[date] += categories_dict[category][date]
+                        else:
+                            month_dict[date] = categories_dict[category][date]
+        return month_dict
+
+    def add_expencies(self):
+        self.update()
+        self.categories_column.append(input())
+        self.date_column.append(date_input())
+        self.expencies_column.append(int_pos_input())
+        self.save()
+
+    def show_all(self):
+        self.update()
```

Comment:
why do we use "update" for read operation ?

Reply:
in case information in the file has been changed manually